### PR TITLE
Handle IFileSystem rename.

### PR DESCRIPTION
**Pull request by Tratcher**
@pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs [hunk 1]
```diff
@@ -157,7 +157,7 @@ protected internal virtual string NormalizePath([NotNull] string path)
             if (path.StartsWith("~\\", StringComparison.Ordinal))
             {
                 // ~\ is not a valid virtual path, and we don't want to replace '\' with '/' as it
-                // ofuscates the error, so just return the original path and throw at a later point
+                // obfuscates the error, so just return the original path and throw at a later point
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs]: :100:

File: test/Microsoft.AspNet.Mvc.Razor.Test/RazorViewEngineOptionsTest.cs [hunk 2]
```diff
@@ -9,13 +9,13 @@ namespace Microsoft.AspNet.Mvc.Razor
     public class RazorViewEngineOptionsTest
     {
         [Fact]
-        public void FileSystemThrows_IfNullIsAsseigned()
+        public void FileProviderThrows_IfNullIsAsseigned()
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Test/RazorViewEngineOptionsTest.cs]: typo: `IfNullIsAssigned`

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs [hunk 3]
```diff
@@ -195,19 +195,17 @@ private static bool IsNetworkPath(string path)
...
             {
-                return FileSystem;
+                return FileProvider;
             }
 
-            // For right now until we can use IWebRootFileSystemProvider, see
-            // https://github.com/aspnet/Hosting/issues/86 for details.
             var hostingEnvironment = requestServices.GetService<IHostingEnvironment>();
-            FileSystem = new PhysicalFileSystem(hostingEnvironment.WebRoot);
+            FileProvider = hostingEnvironment.WebRootFileProvider;
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs]: I'm guessing this is the drop in replacement for what the current code is doing

**Comment by pranavkm** [comment]
Looks good :shipit:

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs [hunk 4]
```diff
@@ -45,18 +45,18 @@ public class MvcRazorHost : RazorEngineHost, IMvcRazorHost
         /// </summary>
         /// <param name="root">The path to the application base.</param>
         public MvcRazorHost(string root) :
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs]: We should probably change the consumer of this to use the new property `hostingEnvironment.WebRootFileProvider`

File: src/Microsoft.AspNet.Mvc.Razor/OptionDescriptors/RazorViewEngineOptions.cs [hunk 5]
```diff
@@ -53,16 +53,16 @@ public TimeSpan ExpirationBeforeCheckingFilesOnDisk
             = new List<ViewLocationExpanderDescriptor>();
 
         /// <summary>
-        /// Gets or sets the <see cref="IFileSystem" /> used by <see cref="RazorViewEngine"/> to locate Razor files on
+        /// Gets or sets the <see cref="IFileProvider" /> used by <see cref="RazorViewEngine"/> to locate Razor files on
         /// disk.
         /// </summary>
         /// <remarks>
-        /// At startup, this is initialized to an instance of <see cref="PhysicalFileSystem"/> that is rooted at the
+        /// At startup, this is initialized to an instance of <see cref="PhysicalFileProvider"/> that is rooted at the
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Razor/OptionDescriptors/RazorViewEngineOptions.cs]: is this comment still true?

**Comment by yishaigalatzer** [comment]
Figure out the couple of minor comments then :shipit:

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs (hunk 4 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs]: The WebRootFileSystemProvider is rooted at `~/wwwroot` versus Views are allowed to be found from `~/` or `~/Views`. Unfortunately this doesn't act as a drop in replacement.

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs (hunk 3 above)
Comment (Tratcher) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs]: Yes

File: src/Microsoft.AspNet.Mvc.Razor/OptionDescriptors/RazorViewEngineOptions.cs (hunk 5 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/OptionDescriptors/RazorViewEngineOptions.cs]: It is (in a round-about way). It gets initialized via the OptionsSetup which gets called the first time you access `IOptionsAccessor<RazorViewEngineOptions>`. Its easier to say Startup instead

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs (hunk 4 above)
Comment (Tratcher) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs]: Views do not come from webroot, they come from app root. See https://github.com/aspnet/Mvc/blob/448ac5a0dc0bc8c53acd4f74df7c533fec6860d7/src/Microsoft.AspNet.Mvc/RazorViewEngineOptionsSetup.cs#L29

**Pull request merged**